api/extension-guides/notebook.md

```diff
@@ -0,0 +1,303 @@
+}
+```
+
+The content provider is then registered in the extension's activation event:
+```ts
+import * as vscode from 'vscode';
+
+export function activate(context: vscode.ExtensionContext) {
+	context.subscriptions.push(
+		vscode.notebook.registerNotebookContentProvider(
+			"ny-notebook-provider", new SampleProvider()
+		)
+	);
+}
+
+class SampleProvider implements vscode.NotebookContentProvider {
+	async openNotebook(uri: vscode.Uri): Promise<vscode.NotebookData> {
+		const content = JSON.parse((await vscode.workspace.fs.readFile(uri)).toString());
+		return {
+			languages: [],
+			metadata: content.metadata,
+			cells: content.cells.map((cell: any) => {
+				if (cell.cell_type === 'markdown') {
+					return {
+						cellKind: vscode.CellKind.Markdown,
+						source: cell.source,
+						language: 'markdown',
+						outputs: [],
+						metadata: {}
+					};
+				} else if (cell.cell_type === 'code') {
+					return {
+						cellKind: vscode.CellKind.Code,
+						source: cell.source,
+						language: content.metadata?.language_info?.name || 'python',
+						outputs: [/* not implemented */],
+						metadata: {}
+					};
+				} else {
+					console.error('Unexpected cell:', cell);
+				}
+			})
+		};
+	}
+
+	// The following are dummy implementations not relevant to this example.
+	onDidChangeNotebook = new vscode.EventEmitter<vscode.NotebookDocumentEditEvent>().event;
+	async resolveNotebook(): Promise<void> { }
+	async saveNotebook(): Promise<void> { }
+	async saveNotebookAs(): Promise<void> { }
+	async backupNotebook(): Promise<vscode.NotebookDocumentBackup> { return { id: '', delete: () => { } }; }
+}
+```
+
+Now try running your extension and opening a Jupyter Notebook formatted file saved with the `.notebook` extension:
+![Notebook showing contents of a Jupyter Notebook formatted file](images/notebook/ipynb-simple-provider.png)
+
+You should be able to open Jupyter-formatted notebooks and view their cells as both plaintext and rendered markdown, as well as edit the cells. However, edits will not be persisted to disk, and you won't be able to run any cells. Saving requires implementing the `saveNotebook`-family commands above, and to run a cell we'll need to implement a `NotebookKernel`.
+
+### Kernel
+[`NotebookKernel` API Reference](/api/references/vscode-api#NotebookKernel)
+
+A `NotebookKernel` is responsible for taking a *code cell* and from it producing some output or set of outputs.
+
+A kernel can either be directly associated with a content provider by setting the `NotebookContentProvider#kernel` property, or registered globally by envoking the `vscode.registerNotebookKernel` function with ...TODO: Figure out how this function is supposed to be called :)...
+
+Samples:
+- [GitHub Issues Notebook](https://github.com/microsoft/vscode-github-issue-notebooks/blob/master/src/notebookProvider.ts): Kernel to execute queries for GitHib Issues
+- [HTTP Request Notebook](): Kernel to issue HTTP requests (TODO: Publish my extension? Where?)
```

Comment:
Or maybe we file PR to the popular file based Rest Client https://marketplace.visualstudio.com/items?itemName=humao.rest-client

Reply:
Thats a good idea, I put up a issue to discuss https://github.com/Huachao/vscode-restclient/issues/628